File: benchmarks/suite/Tak/Tak.rs
fn tak(x: i64, y: i64, z: i64) -> i64 {
    if y < x {
        tak(tak(x - 1, y, z), tak(y - 1, z, x), tak(z - 1, x, y))
    } else {
        z
    }
}

fn main_loop(iters: u64, x: i64, y: i64, z: i64) -> i64 {
    let res = tak(x, y, z);
    if iters == 1 {
        println!("{}", res);
        0
    } else {
        main_loop(iters - 1, x, y, z)
    }
}

fn main() {
    let mut args = std::env::args();
    args.next();
    let iters = args
        .next()
        .expect("Missing Argument iterations")
        .parse::<u64>()
        .expect("Iterations must be a number");
    let x = args
        .next()
        .expect("Missing Argument x")
        .parse::<i64>()
        .expect("x must be a number");
    let y = args
        .next()
        .expect("Missing Argument y")
        .parse::<i64>()
        .expect("y must be a number");
    let z = args
        .next()
        .expect("Missing Argument z")
        .parse::<i64>()
        .expect("z must be a number");
    std::process::exit(main_loop(iters, x, y, z) as i32)
}

File: suite/Tak/Tak.ml
let rec tak x y z = 
  if y<x then 
    tak 
      (tak (x-1) y z) 
      (tak (y-1) z x) 
      (tak (z-1) x y)
  else z

let rec main_loop iters x y z = 
  let res = tak x y z in 
  if iters=1 then 
    print_endline (string_of_int res) 
  else main_loop (iters-1) x y z 

let main = 
  let iters = int_of_string Sys.argv.(1) in 
  let x = int_of_string Sys.argv.(2) in 
  let y = int_of_string Sys.argv.(3) in 
  let z = int_of_string Sys.argv.(4) in 
  main_loop iters x y z

